pkg/apis/serving/v1alpha2/predictor.go

```diff
@@ -79,6 +83,36 @@ func (p *PredictorSpec) Validate(config *InferenceServicesConfig) error {
 	if err := validateReplicas(p.MinReplicas, p.MaxReplicas); err != nil {
 		return err
 	}
+	if err := validateResourceRequirements(predictor.GetResourceRequirements()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func transferToCoreResourceRequirements(rr *v1.ResourceRequirements) *core.ResourceRequirements {
+	resourceRequirements := &core.ResourceRequirements{
+		Limits:   make(core.ResourceList),
+		Requests: make(core.ResourceList),
+	}
+
+	for k, v := range rr.Requests {
+		resourceName := core.ResourceName(string(k))
+		resourceRequirements.Requests[resourceName] = v
+	}
+	for k, v := range rr.Limits {
+		resourceName := core.ResourceName(string(k))
+		resourceRequirements.Limits[resourceName] = v
+	}
+
+	return resourceRequirements
+}
+
+// validate the ResourceRequirements of the model.
+func validateResourceRequirements(rr *v1.ResourceRequirements) error {
```

Comment:
same as above

Reply:
> Nice validation! Can we add the corresponding resource validation unit tests for predictor, transformer, explainer ?

Now in fact all the validation logic is in the API of k8s.io/kubernetes, but sure I can add it if you guys think it's neccessary.

Reply:
from our side we just need to make sure all components get validated for the resources, so we probably just need one simple test case for each component.

Reply:
All the comments addressed, please take a look.